src/provider/index.ts

```diff
@@ -146,9 +153,10 @@ export class Main {
         const appWindow = this._model.getWindow(identity);
         if (appWindow) {
             appWindow.removeIntentListener(payload.intent);
-            return Promise.resolve();
+            return;
         } else {
-            throw new Error('App not found in model');
+            // Previously an Error was thrown at this point. Actually, if for some odd reason the window is not in the model,
```

Comment:
I wouldn't worry about documenting old behaviour in comments, so would get rid of the first sentence.